tests/PySys/mqtt_port/mqtt_port_change_connection_fails/run.py

```diff
@@ -0,0 +1,81 @@
+import sys
+import time
+
+sys.path.append("environments")
+from environment_c8y import EnvironmentC8y
+from pysys.basetest import BaseTest
+
+
+"""
+Validate changing the mqtt port using the tedge command fails without restarting the mqtt server
+
+Given a configured system, that is configured with certificate created and registered in a cloud
+When mqtt.port is set using tedge with sudo
+When the sudo tedge mqtt sub tries to subscribe for a topic and fails to connect to mqtt server
+When the sudo tedge mqtt pub tries to publish a message and fails to connect to mqtt server 
+
+"""
+
+class MqttPortChangeConnectionFails(EnvironmentC8y):
+    def setup(self):
+        self.tedge = "/usr/bin/tedge"
+        self.sudo = "/usr/bin/sudo"
+
+        self.addCleanupFunction(self.mqtt_cleanup)
+
+    def execute(self):
+        # set a new mqtt port for local communication
+        mqtt_port = self.startProcess(
+            command=self.sudo,
+            arguments=[self.tedge, "config", "set", "mqtt.port", "8880"],
+            stdouterr="mqtt_port_set",
+        )
+
+        # validate tedge mqtt pub/sub
+        mqtt_sub = self.startProcess(
+            command=self.sudo,
+            arguments=[self.tedge, "mqtt", "sub", "tedge/measurements"],
+            stdouterr="mqtt_sub",
+            expectedExitStatus="==1",
+            background=True,
+        )
+
+        # publish a message
+        mqtt_pub = self.startProcess(
+            command=self.sudo,
+            arguments=[self.tedge, "mqtt", "pub",
+                       "tedge/measurements", "{ \"temperature\": 25 }"],
+            stdouterr="mqtt_pub",
+            expectedExitStatus="==1",
+        )
+
+        time.sleep(0.5)
+        # wait for a while
+        kill = self.startProcess(
+            command=self.sudo,
+            arguments=["killall", "tedge"],
+            stdouterr="kill_out",
+            expectedExitStatus="==1",
+        )
```

Comment:
With exit status == 1 you are making sure that no process got killed. Better explain this, when this behaviour is intended. the test is probably passing as tedge is exiting after printing the error.

Reply:
Added a comment

Reply:
I fear your comment will confuse the readers. Better add something like "Make sure no tedge binary is running by calling killall tege and checking for exit status 1." If this functionality is intended I would propose to use pidof instead and check the return value there.